test/lifecycle/DestructibleDelayed.test.js

```diff
@@ -0,0 +1,121 @@
+
+require('chai')
+  .use(require('chai-bignumber')(BigNumber))
+  .should();
+
+const twoWeeks = 2 * 7 * 24 * 60 * 60; // In seconds
+
+contract('DestructibleDelayed', function ([owner, anyone]) {
+  let destructibleDelayed;
+
+  describe('Destruction requests', function () {
+    beforeEach(async function () {
+      destructibleDelayed = await DestructibleDelayedMock.new(
+        twoWeeks, { from: owner, value: web3.toWei('10', 'ether') }
+      );
+    });
+
+    it('owner should be able to create a destroyRequest', async function () {
+      await expectEvent.inTransaction(
+        destructibleDelayed.destroyRequest({ from: owner }),
+        'SelfDestructionRequest'
+      );
+
+      const destructionRequested = await destructibleDelayed.destructionRequested();
+      destructionRequested.should.be.equal(true);
+    });
+
+    it('anyone should not be able to create a destroy request', async function () {
+      await expectThrow(
+        destructibleDelayed.destroyRequest({ from: anyone })
+      );
+    });
+
+    context('once destruction requested,', function () {
+      let block;
+      let tx;
+
+      beforeEach(async function () {
+        tx = await destructibleDelayed.destroyRequest({ from: owner });
+        block = await web3.eth.getBlock(tx.receipt.blockNumber);
+      });
+
+      it('destruction time should be set to block.timestamp + destruction_delay', async function () {
+        const destructionTime = await destructibleDelayed.destructionTime();
+        const destructionDelay = await destructibleDelayed.getDestructionDelay();
+
+        destructionTime.should.be.bignumber.equal(
+          destructionDelay.plus(block.timestamp)
+        );
+      });
+
+      it('should not allow a new #destroyRequest call', async function () {
+        await expectThrow(
+          destructibleDelayed.destroyRequest({ from: owner })
+        );
+      });
+
+      it('destroyRequest can be cancelled by owner', async function () {
+        await expectEvent.inTransaction(
+          destructibleDelayed.cancelDestroyRequest({ from: owner }),
+          'SelfDestructionRequestCancelled'
+        );
+
+        const destructionRequested = await destructibleDelayed.destructionRequested();
+
+        destructionRequested.should.be.equal(false);
+      });
+
+      it('anyone should not be able to cancel a destroy request', async function () {
+        await expectThrow(
+          destructibleDelayed.cancelDestroyRequest({ from: anyone })
+        );
+      });
+
+      it('should not allow to call #destroy before delay is passed', async function () {
+        await expectThrow(
+          destructibleDelayed.destroyRequest({ from: owner })
+        );
+      });
+
+      context('once destructionTime is passed', function () {
+        beforeEach(async function () {
+          await increaseTime(twoWeeks + 1);
+        });
+
+        it('should allow #destroy after delay is passed', async function () {
```

Comment:
I think we could merge these last four tests in two:

a) `owner` can destroy and send funds to `anyone` (do check that the exact amount was sent instead of using `greaterThan`)
b) `anyone` cannot destroy

That's nitpicking though, I'll leave it up to you.